velox/exec/Window.h

```diff
@@ -60,13 +60,23 @@ class Window : public Operator {
   }
 
  private:
+  // Used for k preceding/following frames. Index is the column index if k is a
+  // column. value is used to read column values from the column index when k
+  // is a column, and contains the constant value when k is a constant.
+  struct FrameChannelArg {
+    column_index_t index;
+    VectorPtr value;
+  };
+
   // Structure for the window frame for each function.
   struct WindowFrame {
     const core::WindowNode::WindowType type;
     const core::WindowNode::BoundType startType;
     const core::WindowNode::BoundType endType;
-    const std::optional<column_index_t> startChannel;
-    const std::optional<column_index_t> endChannel;
+    // These parameters are set only for k PRECEDING and k FOLLOWING frame
```

Comment:
It might be clearer to say 

```
// Set only when startType is BoundType::kPreceding or kFollowing.
const std::optional<FrameChannelArg> start;
// Set only when endType is BoundType::kPreceding or kFollowing.
const std::optional<FrameChannelArg> end;
```

Also, consider renaming xxxFrameArg to xxx since Frame is implied by the struct name.